test/dotnet-new3.UnitTests/Approvals/DotnetNewHelp.CanShowHelpForTemplate_Classlib.approved.txt

```diff
@@ -1,22 +1,35 @@
 ﻿Class Library (C#)
 Author: Microsoft
 Description: A project for creating a class library that targets .NET Standard or .NET Core
-Options:                                                                             
-  -f|--framework  The target framework for the project.                              
-                      net7.0            - Target net7.0                              
-                      netstandard2.1    - Target netstandard2.1                      
-                      netstandard2.0    - Target netstandard2.0                      
-                      net5.0            - Target net5.0                              
-                      netcoreapp3.1     - Target netcoreapp3.1                       
-                  Default: net7.0                                                    
 
-  --langVersion   Sets the LangVersion property in the created project file          
-                  text - Optional                                                    
+Usage:
+  new3 classlib [options] [template options]
 
-  --no-restore    If specified, skips the automatic restore of the project on create.
-                  bool - Optional                                                    
-                  Default: false                                                     
+Options:
+  -n, --name <name>       The name for the output being created. If no name is specified, the name of the output directory is used.
+  -o, --output <output>   Location to place the generated output.
+  --dry-run               Displays a summary of what would happen if the given command line were run if it would result in a template creation.
+  --force                 Forces content to be generated even if it would change existing files.
+  --no-update-check       Disables checking for the template package updates when instantiating a template.
+  -lang, --language <C#>  Specifies the template language to instantiate.
+  --type <project>        Specifies the template type to instantiate.
 
+Template options:
+  -f, --framework <net5.0|net7.0|netcoreapp3.1|netstandard2.0|netstandard2.1>  The target framework for the project.
```

Comment:
@baronfel how do you feel about this notation? or should we rather not list possible options for choices?

now it makes 1st column really large

Reply:
also, for required arguments: the syntax of first column is `--my-choice <val1|val2|val3|val4> required`.

Reply:
I've been doing some digging around that, trying out sample code for popular golang/nodejs/python CLI libraries to compare. So far they either:

* put the choices in a longer documentation section (like you've shown with the TFM documentation section),
* after a certain line width, put the documentation on a new line and ignore the long line for purposes of indentation on other lines, or 
* sidestep the issue entirely by forcing users to write all usage/help documentation manually

of those options, in an ideal world I'd like to have a behavior like:

* start by doing the current inline behavior
* if the line gets too long (we can debate what too long means, but let's say ~50 of terminal width), put the documentation on a new line compared to the flag usage example

or a behavior like:
* if there are more than N choices, do not show the choices in line with the argument. instead, just rely on the full listing of the options in the help documentation.

I'm not sure how feasible something like that is in the current S.CL API, partially because at least part of that seems like a terminal concern, not a CLI app concern necessarily. This is where something like GNU readline would be useful.